x-pack/plugins/maps/public/shared/layers/sources/wms_source/wms_source.js

```diff
@@ -73,70 +69,3 @@ export class WMSSource extends AbstractTMSSource {
     return `${this._descriptor.serviceUrl}?bbox={bbox-epsg-3857}&format=image/png&service=WMS&version=1.1.1&request=GetMap&srs=EPSG:3857&transparent=true&width=256&height=256&layers=${this._descriptor.layers}&styles=${styles}`;
```

Comment:
In some WMS service, users can define extra query options to the WMS server as query parameters. For example, [this request](https://kerdoc.cica.es/gs/geoserver/gcm/ows?SERVICE=WMS&CQL_FILTER=agno%3D2019%20and%20mes%3D1&bbox=-20037508.342789244,10018754.17139462,-10018754.171394622,20037508.342789244&format=image/png&service=WMS&version=1.3.0&request=GetMap&srs=EPSG:3857&transparent=true&width=256&height=256&layers=anomalias_pre_porcentaje_espacial) sets the `CQL_FILTER` query parameter to `agno=2019 and mes=1` to filter the result to January 2019. 

It would be nice if we could allow URL encoded query parameters in the URL field (e.g. https://kerdoc.cica.es/gs/geoserver/gcm/ows?CQL_FILTER=agno%3D2018%20and%20mes%3D1) and not overwrite them. Maybe we can re-use some of the functionality [here](https://github.com/elastic/kibana/blob/master/src/legacy/ui/public/utils/query_string.js#L89-L112)

If this is out of scope of this PR, let me know and I will open a new issue.

Reply:
Lets create another issue for this and keep this PR just focused on GetCapabilities